src/Bicep.Core/Syntax/ImportSpecification.cs

```diff
@@ -24,18 +24,21 @@ public class ImportSpecification : ISymbolNameSource
         // language=regex
         private const string SemanticVersionPattern = @"(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?";
 
-        private static readonly Regex SpecificationPattern = new(
-            @$"^(?<name>{NamePattern})@(?<version>{SemanticVersionPattern})$",
-            RegexOptions.ECMAScript | RegexOptions.Compiled);
+        private static readonly Regex BareSpecification = new(
+            @$"(?<name>{NamePattern})@(?<version>{SemanticVersionPattern})$");
```

Comment:
Shouldn't we keep the leading `^`? (in regex, this means "only match from the beginning of the string")

Reply:
No, in this case I am deliberately capturing only the last `NamePattern`, this is so that I can get the implicit symbol for the provider.